eth_defi/token_tax.py

```
@@ -0,0 +1,134 @@
+    base_token = base_token_details.contract
+
+    # approve router to spend tokens
+    quote_token.functions.approve(router.address, quote_token_details.convert_to_raw(99999)).transact({"from": buy_account})
+
+    path = [quote_token.address, base_token.address]
+    amountIn = quote_token_details.convert_to_raw(1)
+    # Figure out base_token/quote_token trading pair
+    initial_base_bal = base_token.functions.balanceOf(buy_account).call()
+    # Buy base_token with buy_account
+    router.functions.swapExactTokensForTokensSupportingFeeOnTransferTokens(
+        amountIn,
+        0,
+        path,
+        buy_account,
+        FOREVER_DEADLINE
+    ).transact({"from": buy_account})
+
+    received_amt = base_token.functions.balanceOf(buy_account).call() - initial_base_bal
+    uniswap_price = router.functions.getAmountsOut(amountIn, path).call()[1]
+
+    # Measure the loss as "buy tax"
+    buy_tax_percent = (uniswap_price - received_amt) / uniswap_price
+
+    # Transfer tokens to sell_account
+    # Measure the loss as "transfer tax"
+    base_token.functions.transfer(sell_account, received_amt).transact({"from": buy_account})
+
+    received_amt_by_seller = base_token.functions.balanceOf(sell_account).call()
+
+    transfer_tax_percent = (received_amt - received_amt_by_seller) / received_amt
+
+    # Sell tokens
+    base_token.functions.approve(router.address, received_amt_by_seller).transact({"from": sell_account})
+    path = [ base_token.address, quote_token.address]
+
+    sell_tax = 0
+    sell_tax_percent = 0
+    try:
+        # this method will revert in case of low liquidity of the token
+        router.functions.swapExactTokensForTokensSupportingFeeOnTransferTokens(
+            received_amt_by_seller,
+            0,
+            path,
+            sell_account,
+            FOREVER_DEADLINE
+    ).transact({"from": sell_account})
+    except:
+        print("Low liquidity. Sell method failed")
```

Exceptions must not be swallowed, otherwise the called of the function do not know what happened. Create a proper exception class, document it and raise it here.